tests/multigrid-global-coarsening/mg_transfer_util.h

```diff
@@ -167,4 +172,138 @@ test_transfer_operator(
   }
 }
 
+
+
+template <int dim, typename Number, typename MeshType>
+void
+test_non_nested_transfer(
+  const MGTwoLevelTransferNonNested<dim,
+                                    LinearAlgebra::distributed::Vector<Number>>
+    &                          transfer,
+  const MeshType &             dof_handler_fine,
+  const MeshType &             dof_handler_coarse,
+  const Function<dim, Number> &function =
+    Functions::ZeroFunction<dim, Number>(),
+  const unsigned int mg_level_fine   = numbers::invalid_unsigned_int,
+  const unsigned int mg_level_coarse = numbers::invalid_unsigned_int)
+{
+  AffineConstraints<Number> constraint_fine;
+  DoFTools::make_hanging_node_constraints(dof_handler_fine, constraint_fine);
+  constraint_fine.close();
+
+  // perform prolongation
+  LinearAlgebra::distributed::Vector<Number> src, dst;
+
+  initialize_dof_vector(dst, dof_handler_fine, mg_level_fine);
+  initialize_dof_vector(src, dof_handler_coarse, mg_level_coarse);
+
+  // test prolongation
+  {
+    src = 0.0;
+    VectorTools::interpolate(dof_handler_coarse, function, src); // src = 1.0
+    dst = 0.0;
+    transfer.prolongate_and_add(dst, src);
+
+    // transfer operator sets only non-constrained dofs -> update the rest
+    // via constraint matrix
+    constraint_fine.distribute(dst);
+
+    // Visualize the result
+    {
+      DataOut<2> data_out;
+      data_out.attach_dof_handler(dof_handler_coarse);
+      data_out.add_data_vector(
+        src,
+        "coarse_solution",
+        DataOut_DoFData<dim, dim>::DataVectorType::type_dof_data);
+      data_out.build_patches();
+      std::ofstream output_coarse("coarse_sol.vtk");
+      data_out.write_vtk(output_coarse);
+      data_out.clear();
+      std::ofstream output_fine("prolonged_sol.vtk");
+      data_out.attach_dof_handler(dof_handler_fine);
+      data_out.add_data_vector(
+        dst,
+        "prolonged_solution",
+        DataOut_DoFData<dim, dim>::DataVectorType::type_dof_data);
+
+      data_out.build_patches();
+      data_out.write_vtk(output_fine);
+    }
```

Comment:
Do we really want to write the result? Should we check it? As tests are run on systems with limited disk performance, I would prefer not writing the data.